packages/cli/src/util/extension/exec.ts

```diff
@@ -0,0 +1,91 @@
+import { dirname } from 'path';
+import execa from 'execa';
+import listen from 'async-listen';
+import { scanParentDirs, walkParentDirs } from '@vercel/build-utils';
+import { createProxy } from './proxy';
+import type Client from '../client';
+import which from 'which';
+
+/**
+ * Attempts to execute a Vercel CLI Extension.
+ *
+ * If the extension was found and executed, then the
+ * exit code is returned.
+ *
+ * If the program could not be found, then an `ENOENT`
+ * error is thrown.
+ */
+export async function execExtension(
+  client: Client,
+  name: string,
+  args: string[],
+  cwd: string,
+  apiUrl: string,
+  token?: string
+): Promise<number> {
+  const { debug } = client.output;
+  const extensionCommand = `vercel-${name}`;
```

Comment:
Is it OK if we use `vercel-cli-`? I think that will reduce the risk of collisions in future.

For now I'll use that in our package, but we can remove the `cli-` part if you feel the name-spacing isn't worth the effort. We could also use `vercel-ext-` or `vc-*-`.

Reply:
Is it really necessary though? I don't really see any risk of a collision. If there's a `vercel-foo` command that exists out in the wild today, it would still work just fine as an extension (just wouldn't be relying on any extension features) or as standalone.

Reply:
It would work fine, but may not have been intended to be an extension - which may be an issue in future if we're doing more with extensions, like providing more context, etc.

I don't think it hurts to be safe here, but I understand that it may feel unneeded. I'll defer to you and the CLI team to make the call on this.